segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentContainer.java

```diff
@@ -736,30 +736,34 @@ public CompletableFuture<List<ExtendedChunkInfo>> getExtendedChunkInfo(String st
         return wrapper.getExtendedChunkInfoList(streamSegmentName, true);
     }
 
+    @SneakyThrows
     @Override
     public CompletableFuture<Void> checkChunkStorageSanity(int containerId, String chunkName, int dataSize, Duration timeout) {
         val chunkedSegmentStorage = (ChunkedSegmentStorage) storage;
         UtilsWrapper wrapper = new UtilsWrapper(chunkedSegmentStorage, BUFFER_SIZE, timeout);
         return wrapper.checkChunkSegmentStorageSanity(chunkName, 10);
     }
 
+    @SneakyThrows
     @Override
     public CompletableFuture<Void> evictMetaDataCache(int containerId, Duration timeout) {
         val chunkedSegmentStorage = (ChunkedSegmentStorage) storage;
         UtilsWrapper wrapper = new UtilsWrapper(chunkedSegmentStorage, BUFFER_SIZE, timeout);
         return wrapper.evictMetadataCache();
     }
 
+    @SneakyThrows
     @Override
     public CompletableFuture<Void> evictReadIndexCache(int containerId, Duration timeout) {
         val chunkedSegmentStorage = (ChunkedSegmentStorage) storage;
         UtilsWrapper wrapper = new UtilsWrapper(chunkedSegmentStorage, BUFFER_SIZE, timeout);
         return wrapper.evictReadIndexCache();
     }
 
+    @SneakyThrows
     @Override
     public CompletableFuture<Void> evictReadIndexCacheForSegment(int containerId, String segmentName, Duration timeout) {
-        val chunkedSegmentStorage = (ChunkedSegmentStorage) storage;
+        val chunkedSegmentStorage = ChunkedSegmentStorage.getReference(this.storage);
```

Comment:
very good catch.
`val chunkedSegmentStorage = ChunkedSegmentStorage.getReference(this.storage);`
Please do the same for other 3 methods you have added

Reply:
Sure.